### ISSUE TITLE
dropdown menu reusable interaction pattern

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1615]
Whenever a toggle + menu interaction pattern is implemented in a component, the methods used to handle user interaction should be consistent across all the components that use this pattern. 

This issue is to fully implement this interaction pattern for one of the components.
Then follow-up issues should capture this work for any other components that include this pattern.

Some of the components we have now that include this interaction pattern are:
- Dropdown
- Select
- Pagination
- Context Selector
- Wizard
- AppLauncher

**[Details about interaction pattern requirements](https://docs.google.com/document/d/1YdjO23kgB4MKmc7_rvXbS5ml0a08FobGDGQPYeTPEgo/edit?usp=sharing)**

This work is dependent on having more modular components for menu and toggle available in core - https://github.com/patternfly/patternfly-next/issues/1429

**Comment by nicolethoen** [comment]
This can be considered a duplicate of https://github.com/patternfly/patternfly-react/issues/6014 since all dropdown menus will use the same Menu component under the hood, they will be using the same interaction patterns.
